client/app/_api/createUserHouseApi.ts

```diff
@@ -4,8 +4,12 @@ import { UserHouse } from '@/_types'
 export const createUserHouseApi = async (
   houseId: string,
 ): Promise<UserHouse> => {
-  const response = await apiClient.post<UserHouse>('/user-houses', {
-    houseId,
-  })
-  return response.data
+  return apiClient
+    .post<UserHouse>('/user-houses', {
+      houseId,
+    })
+    .then((response) => response.data)
+    .catch((error) => {
+      throw error
+    })
```

Comment:
Can I ask why you changed from using `await`? `await` would also throw an error if the POST request fails.

Reply:
Well, I struglled to catch the error using React Boundary or with `isError` in this part but I ended up to use just async function inside of `useEffect` so I didn't have to change here.. I changed back! 

https://github.com/HAK2024/fairy-share/pull/89/commits/e993ba3136e295035e91921e15e432f32a92e44a